backend/dal/Repositories/ResearchFileRepository.cs

```diff
@@ -46,8 +65,23 @@ public PimsResearchFile Add(PimsResearchFile researchFile)
         }
 
         /// <summary>
-        /// Get a page with an array of leases within the specified filters.
-        /// Note that the 'researchFilter' will control the 'page' and 'quantity'.
+        /// Update the specified research file.
+        /// </summary>
+        /// <param name="researchFile"></param>
+        /// <returns></returns>
+        public PimsResearchFile Update(PimsResearchFile researchFile)
+        {
+            researchFile.ThrowIfNull(nameof(researchFile));
+
+            researchFile.RfileNumber = GenerateRFileNumber();
+
+            this.Context.PimsResearchFiles.Update(researchFile);
```

Comment:
I'm always worried about doing this. I believe this updates the db with the passed research file, but also updates all child relationships. For example, if this research file entity contained a number of changes to PimsProperty entities that would change those as well. Probably fine but we need to be careful about possible inadvertent updates.